pkg/scale/scale.go

```diff
@@ -103,51 +103,136 @@ type cruiseControlScaler struct {
 	client *client.Client
 }
 
-// Status returns a CruiseControlStatus describing the internal state of Cruise Control.
-func (cc *cruiseControlScaler) Status(ctx context.Context) (CruiseControlStatus, error) {
+// Status returns a StatusTaskResult describing the internal state of Cruise Control.
+func (cc *cruiseControlScaler) Status(ctx context.Context) (StatusTaskResult, error) {
 	req := api.StateRequestWithDefaults()
 	req.Verbose = true
 	resp, err := cc.client.State(ctx, req)
 	if err != nil {
-		return CruiseControlStatus{}, err
+		return StatusTaskResult{}, err
+	}
+
+	// if the execution takes too much time, then Cruise Control converts the request into async
+	// and returns the taskID
+	if resp.Result == nil {
+		return StatusTaskResult{
+			TaskResult: &Result{
+				TaskID:             resp.TaskID,
+				StartedAt:          resp.Date,
+				ResponseStatusCode: resp.StatusCode,
+				RequestURL:         resp.RequestURL,
+				State:              v1beta1.CruiseControlTaskActive,
```

Comment:
At this point the State should be v1beta1.CruiseControlTaskInExecution because CC is doing the operation in the background

Reply:
Changed in https://github.com/banzaicloud/koperator/pull/974/commits/2c47d7dbbae5a00806acb152d172b3ad4ada2b42